eventuals/scheduler.h

```diff
@@ -37,12 +37,44 @@ class Scheduler {
       return previous;
     }
 
-    Context(Scheduler* scheduler)
-      : scheduler_(scheduler) {}
+    auto* scheduler() {
+      return scheduler_;
+    }
 
-    Context(Context&& that) = delete;
+    const std::string& name() {
+      return name_;
+    }
+
+    Context(
+        Scheduler* scheduler,
+        const std::string& name = "default",
+        void* data = nullptr,
+        Context* parent = nullptr)
+      : data_(data),
+        parent_(parent),
+        scheduler_(scheduler),
+        name_(name) {}
```

Comment:
See other comments, but in here let's do:
```suggestion
        name_(name) {
          scheduler_->Clone(this);
        }
```